docs/proposals/DNSPolicy.md

```diff
@@ -0,0 +1,409 @@
+- Define how health checks should work, this will be part of a separate proposal
+
+
+## Terms
+
+- **managed listener**: This is a listener with a host backed by a DNS zone managed by the multi-cluster gateway controller
+- **hub cluster**: control plane cluster that managed 1 or more spokes
+- **spoke cluster**: a cluster managed by the hub control plane cluster. This is where gateway are instantiated
+
+## Proposal
+
+Provide a control plane DNSPolicy API that uses the idea of direct [policy attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment) from gateway API that allows a load balancing strategy and TTLs to be applied to the DNS records for any managed listeners being served by the data plane instances of this gateway. 
+The DNSPolicy also covers DNS based health checks but that is not covered in this document.
+
+Below is a draft API for what we anticipate the DNSPolicy to look like
+
+```yaml 
+
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+spec:
+  targetRef: # defaults to gateway gvk and currrent namespace
+    name: gateway-name
+  health:
+   ...
+  ttl: 
+    value: 60
+    gracePeriod: 600 #(seconds to wait beyond TTL before allowing deletion of gateway) 
+  loadBalancing:
+     #one or both of geo and weighted are required 
+    weighted:
+     default: 10  #required if weighted is chosen
+     custom:
+     - attribute: cloud  #optional with both geo and weighted
+       value: AWS
+       weight: 10
+     - attribute: cloud
+       value: GCP
+       weight: 20
+    geo:
+      default: IE # required if geo is chosen
+```  
+
+With the above API the two strategies are weighted or geo however you can use both to get a geo and weighted within a geo load balancing. By defining a geo section, you are indicating that you want to use a geo based strategy. A geo based strategy can be complimented with a weighted strategy. If you do not define a geo section, then you are choosing a purely weighted strategy.
+  
+### Load Balancing Strategies  
+
+Geo and Weighted load balancing are well understood strategies and this API effectively allow a complex requirement to be expressed relatively simply and executed by the gateway controller. Our default policy will execute a "Round Robin" weighted strategy which reflects the current behaviour.
+
+```yaml 
+
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+name: default-round-robin
+spec:
+  targetRef: # defaults to gateway gvk and currrent namespace
+    name: gateway-name
+  health:
+   ...
+  ttl: 
+    value: 60
+    gracePeriod: 240 #(seconds to wait beyond TTL before allowing deletion of gateway) 
+  loadBalancing:
+     #one or both of geo and weighted are required 
+    weighted:
+     default: 10  #required if weighted
```

Comment:
since everything is always weighted, should this just be required?

Reply:
Yeah I think I agree with this